### ISSUE TITLE
m_validSamples too low

### ISSUE THREAD

**Issue by Marcin-Fenger** [schreibfaul1/ESP32-audioI2S#915]
I print m_validSamples in audio_process_i2s function. After last changes m_validSamples are dramaticaly low in playChunk. In previous versions was 1024 or 1152 depend on codec. Now is 55 or 88 or 97 and it is not stable. I make FFT with core 0. Why now is so low?

**Comment by schreibfaul1** [comment]
Hello Marcin,
that is not the case. The number of samples is determined by the size of the frames, which is usually between 1 and 2KB, larger for wav and mp3.

I think what you are observing is the effect that not all samples are always copied to the I2S DMA. Then there is a second attempt to write the remaining samples.

`i2s_channel_write(m_i2s_tx_handle, (int16_t*)m_outBuff + count, validSamples * sampleSize, &i2s_bytesConsumed, 5);`
waits 5ms, if the I2S buffer is still full by then, the number of samples not written is calculated. In earlier versions, samplesize was not calculated correctly for mono, perhaps this is the reason. If you want as many samples as possible to be written, you can increase the "timeout_ms" from 5 to 20.

**Comment by Marcin-Fenger** [comment]
Hello

Very very thank You. I have increased timeout to 20ms. Now everythig is all right and stable as rock. I use ESP 32 WROVER IE n16r8. After this change I fill better quality of 1ch stations. Think i should be changed to higher value for idf4.4. In previous settings (timeout 10) 1ch stations start play jammed sound after few minutes of start and return to normal play and again. I set 20ms and now is cool. Gold tip! Thanks.

BTW line 5730   if(m_f_Log) log_i("chunksize %d", chunksize); spam serial log with chunksize on serial. Reconsider comment it.

**Comment by schreibfaul1** [comment]
Yes, increasing it from 5ms to 10ms seems to be better. This often saves a second run.
